test/integration/utility.h

```diff
@@ -62,32 +62,45 @@ using BufferingStreamDecoderPtr = std::unique_ptr<BufferingStreamDecoder>;
 class RawConnectionDriver {
 public:
   using ReadCallback = std::function<void(Network::ClientConnection&, const Buffer::Instance&)>;
+  using ExitableReadCallback =
+      std::function<bool(Network::ClientConnection&, const Buffer::Instance&)>;
 
-  RawConnectionDriver(uint32_t port, Buffer::Instance& initial_data, ReadCallback data_callback,
+  RawConnectionDriver(uint32_t dst_port, Buffer::Instance& initial_data, ReadCallback data_callback,
```

Comment:
Why are all of these changes needed? I'm not sure why you are using "raw HTTP" for these tests?

Reply:
The test is to prove some connections survive the listener update. Any suggestion?

Reply:
You can do this with the normal fake HTTP infra vs. any changes to the "raw" stuff. You can just make a connection/request, do your thing, and then make another connection/request. You can also wait for disconnect/reset/drain on ones that are draining. Let's revert all of these changes as I don't think they should be necessary to test your feature.